packages/core/mesh/messaging/src/signal-client/signal-client.ts

```diff
@@ -65,7 +55,7 @@ export type CommandTrace = {
  * Establishes a websocket connection to signal server and provides RPC methods.
  */
 export class SignalClient implements SignalMethods {
-  private _state = SignalState.CLOSED;
+  private readonly _pushState = new PushStream<SignalState>();
```

Comment:
not sure about observables in core, on one hand coupling events with state is nice, but on the other hand the flexibility of having separate events feels more important.

Reply:
although this example does look better with observables (its just one value coupled with it's update event), selectively using observables in core packages might add confusion on whether a specific case should be handled by events or observables. in this case for example, we might want to replace `stateChanged` with `connected` `closed`, and `error` events, which would be hard with observables

Reply:
overall, obserables are a great fit in the client sdk, where the need to make api consistent and bulletproof is higher. In lower-level code I think flexibility and granularity is more important, and I'd prefer to stick to events

Reply:
Yesterday I thought it was going to be necessary or at least make it far easier to hook up for some reason, but that didn't turn out to be the case. I think the implementations are almost identical in this case, so yeah, I'll just switch it back.